src/transfer.cpp

```diff
@@ -356,29 +356,78 @@ void Transfer::failed(error e, dstime timeleft)
 
     if (e != API_EBUSINESSPASTDUE)
     {
-        if (e != API_EOVERQUOTA)
+        if (e == API_EOVERQUOTA)
+        {
+            if (!slot)
+            {
+                client->activateoverquota(timeleft);
+                bt.backoff(timeleft ? timeleft : NEVER);
+                client->app->transfer_failed(this, e, timeleft);
+                ++client->performanceStats.transferTempErrors;
+            }
+            else
+            {
+                bool allForeignTargets = true;
+                for (file_list::iterator it = files.begin(); it != files.end(); it++)
+                {
+                    Node *node = client->nodebyhandle((*it)->h);
+                    if (node)
+                    {
+                        handle rootnode = client->getrootnode(node)->nodehandle;
+                        if (client->rootnodes[0] == rootnode)
```

Comment:
Probably we should compare against all 3 root nodes here, as it's technically possible to upload/download from //bin and //in node trees.    Perhaps we should have a function `bool MegaClient::isAccountRootNode(handle)` to make that check easy?  thanks

Reply:
You are right, I've created the method isAccountRootNode to check it.